In [2]:
import pandas as pd

hotel_project = pd.read_csv('hotel_bookings.csv')
hotel_cancel = hotel_project[hotel_project['is_canceled']>0]

In [25]:
parkingi = hotel_cancel.groupby(['required_car_parking_spaces']).agg({'required_car_parking_spaces':'count'})
parkingi = parkingi.rename(columns={'required_car_parking_spaces':'count'})
parkingi

count
required_car_parking_spaces       
0                            44224

In [5]:
hotel_project['required_car_parking_spaces'].max()  # rezerwacja z najwyzsza iloscia zadanych miejsc parkingowych

8

In [22]:
procent = hotel_project.groupby(['required_car_parking_spaces']).agg({'required_car_parking_spaces':'count'})
procent = procent.rename(columns={'required_car_parking_spaces':'count'})
procent

count
required_car_parking_spaces        
0                            111974
1                              7383
2                                28
3                                 3
8                                 2

In [78]:
procent['%'] = round((procent['count']/(procent['count']).sum())*100,2)
procent

count      %
required_car_parking_spaces               
0                            111974  93.79
1                              7383   6.18
2                                28   0.02
3                                 3   0.00
8                                 2   0.00

In [75]:
stosunek = round((parkingi['count']/procent['count'])*100,2)
stosunek = pd.DataFrame(stosunek)
stosunek = stosunek.rename(columns={'count':'%'})
stosunek


%
required_car_parking_spaces       
0                            39.49
1                              NaN
2                              NaN
3                              NaN
8                              NaN

In [76]:
x = (0.395*0.2)*100 
x

7.900000000000001

Zgodnie z powyższymi wyliczeniami, 44224 odwolane rezerwacje, stanowia 39,5% wszystkich rezerwacji. Odwolania te dotyczą tylko i wyłącznie miejsc parkingowych = 0. 
Zgodnie z tablica korelacji, 'required_car_parking_spaces' to waga korelacji 20%.

Zatem łączna waga parkingów to 7,9%. 

Dla x > 50% przyjmujemy, że klient odwola rezerwacje (1)
Dla x < 50% przyjmujemy, że klient nie odwola rezerwacji (0).

Elementy wplywajace na IS Cancelled wedlug rankingu korelacji
1. lead_time - 29% waga - Piotr
2. total_of_special_requests  - 23% waga - Iwona
3. required_car_parking_spaces  - 20% waga - Julita
4. booking_changes - 14% waga - Adam
5. previous_cancellations  - 11% waga - Best effort basis *
6. is_repeated_guest  - 8% waga - Best effort basis *

In [77]:
def parking():
    # zakladamy, ze hotel dysponuje ograniczona liczba miejsc. Przyjmujemy, ze jest to 8 (max liczba zadanych miejsc parkingowych w zestawieniu - nie zostaly odwolane).
    ile_miejsc = int(input('Podaj liczbę miejsc parkingowych, zarezerwowanych przez klienta. Max miejsc: 8'))
    if ile_miejsc in range(0,9): 
        if ile_miejsc == 0:  
            return 0   # odwolanie rezerwacji
        else:
            return 0    # nieodwolanie rezerwacji
    elif ile_miejsc > 8:
        print('Nie ma tyle miejsc. Sprobuj ponownie')
        ile_miejsc = int(input('Podaj liczbę miejsc parkingowych, zarezerwowanych przez klienta. Max miejsc: 8'))
            

parking()

0